cabby/data/extract.py

```diff
@@ -14,140 +14,197 @@
+  backlinks_pages = wpq.get_backlinks_items_from_wikipedia_titles(
+    wikipedia_titles)
+  backlinks_items = []
+  for list_backlinks in backlinks_pages:
+    backlinks_items.append(
+      [wpi.WikipediaEntity.from_backlinks_api_result(result) for
+        result in list_backlinks])
+
+  # Change backlinks pages to Geodata dataset format.
+  for list_backlinks, original_wikipedia, original_wikidata in \
+      zip(backlinks_items, wikipedia_items, wikidata_items):
+    for backlink in list_backlinks:
+      wikigeo_sample = wikigeo.WikigeoEntity.from_wiki_items(
+        backlink, original_wikipedia, original_wikidata).sample
+
+      sample_text = wikigeo_sample['text']
+      if any(dict['text'] == sample_text for dict in geo_data):
+        continue
+      geo_data.append(wikigeo_sample)
+  return geo_data
 
 
 def get_data_by_qid(qid: Text) -> Sequence:
-    '''Get data from Wikipedia and Wikidata by region" 
-    Arguments:
-        qid(Text): The qid of the Wikidata to extract items from.
-    Returns:
-        The Wikipedia (text,title) and Wikidata (location) data found.
-    '''
+  '''Get data from Wikipedia and Wikidata by region" 
+  Arguments:
+    qid(Text): The qid of the Wikidata to extract items from.
+  Returns:
+    The Wikipedia (text,title) and Wikidata (location) data found.
+  '''
 
-    # Get Wikidata items by region.
-    wikidata_results = wdq.get_geofenced_wikidata_items_by_qid(qid)
-    wikidata_items = [wdi.WikidataEntity.from_sparql_result(result)
-                      for result in wikidata_results]
+  # Get Wikidata items by region.
+  wikidata_results = wdq.get_geofenced_wikidata_items_by_qid(qid)
+  wikidata_items = [wdi.WikidataEntity.from_sparql_result(result)
+            for result in wikidata_results]
 
-    return get_wikigeo_data(wikidata_items)
+  return get_wikigeo_data(wikidata_items)
 
 
 def get_data_by_region(region: Text) -> Sequence:
-    '''Get data from Wikipedia and Wikidata by region" 
-    Arguments:
-        region(Text): The region to extract items from.
-    Returns:
-        The Wikipedia (text,title) and Wikidata (location) data found.
-    '''
-
-    # Get Wikidata items by region.
-    wikidata_results = wdq.get_geofenced_wikidata_items(region)
-    wikidata_items = [wdi.WikidataEntity.from_sparql_result(result)
-                      for result in wikidata_results]
-
-    return get_wikigeo_data(wikidata_items)
-
-
-def split_dataset( 
+  '''Get data from Wikipedia and Wikidata by region" 
+  Arguments:
+    region(Text): The region to extract items from.
+  Returns:
+    The Wikipedia (text,title) and Wikidata (location) data found.
+  '''
+
+  # Get Wikidata items by region.
+  wikidata_results = wdq.get_geofenced_wikidata_items(region)
+  wikidata_items = [wdi.WikidataEntity.from_sparql_result(result)
+            for result in wikidata_results]
+
+  return get_wikigeo_data(wikidata_items)
+
+
+def get_data_by_region_expansive(region: Text, path_osm: Text) -> Sequence:
```

Comment:
what does expansive mean?
looking at the code it seems like it is returning the region given the specific region boundaries. may be rename the function as get_data_by_region_boundaries?

Reply:
may be rename as get_data_by_region_with_osm?
'expansive' is not very evident